### PR TITLE
bugfix for tx selection: 

### PR DESCRIPTION
**Pull request by eranrund**
Prior to this change full-service would select the smallest TXOs, even though all previous checks look at the large ones.
This results in incorrect txo selection (total sum of selected txos would be less than desired amount), even when the account has sufficient txos.

This should've been caught by unit tests, so a unit test still needs to be written to cover this.

### CODE REVIEW

File: full-service/src/db/txo.rs [hunk 1]
```diff
@@ -1847,11 +1848,156 @@ mod tests {
...
+            None,
+            &wallet_db.get_conn().unwrap(),
+        );
+        assert!(result.is_err());
+
+        // testing over the max spendable but under the total in account, should fail
+        let result = Txo::select_unspent_txos_for_value(
+            &account_id.to_string(),
+            201 as u64,
+            None,
```
Comment (eranrund) [inline full-service/src/db/txo.rs:1896]: This doesn't match the comment above (max spendable is None here)

File: full-service/src/db/txo.rs [hunk 2]
```diff
@@ -1847,11 +1848,156 @@ mod tests {
...
+            &wallet_db.get_conn().unwrap(),
+        );
+        assert!(result.is_err());
+
+        // testing at the max spendable limit, should pass and have 16 txouts selected
+        let result = Txo::select_unspent_txos_for_value(
+            &account_id.to_string(),
+            200 as u64,
+            None,
```
Comment (eranrund) [inline full-service/src/db/txo.rs:1905]: Max spendable not being set

File: full-service/src/db/txo.rs [hunk 3]
```diff
@@ -1847,11 +1848,156 @@ mod tests {
...
+            &wallet_db.get_conn().unwrap(),
+        )
+        .unwrap();
+        assert_eq!(result.len(), 16);
+        let sum = result
+            .into_iter()
+            .map(|x| x.value)
+            .reduce(|a, b| a + b)
```
Comment (eranrund) [inline full-service/src/db/txo.rs:1913]: Isnt this `.sum()`?

File: full-service/src/db/txo.rs [hunk 4]
```diff
@@ -1847,11 +1848,156 @@ mod tests {
...
+            .unwrap();
+        assert_eq!(3 as i64, sum);
+    }
+
+    #[test_with_logger]
+    fn test_eran_issue(logger: Logger) {
```
Comment (eranrund) [inline full-service/src/db/txo.rs:1935]: Interesting name... :p

File: full-service/src/db/txo.rs [hunk 5]
```diff
@@ -1847,11 +1848,156 @@ mod tests {
...
+            src_txos.push(txo);
+        }
+
+        for i in 1..=500 {
+            let (_txo_id, txo, _key_image) =
+                create_test_received_txo(&account_key, i, 100000000, i, &mut rng, &wallet_db);
+            src_txos.push(txo);
+        }
+
+        // testing over the total in account, should fail
```
Comment (eranrund) [inline full-service/src/db/txo.rs:1980]: This doesn't match what the test is testing for

**Review by eranrund — COMMENTED** [review]
Thank you for dealing with all the flaky tests!
I cant select "Request changes" because its my own PR, but there are a few places where the test description doesn't match whats actually being tested.

File: full-service/src/db/txo.rs (hunk 3 above)
Comment (BrianCorbin) [reply full-service/src/db/txo.rs:1913]: O_O

File: full-service/src/db/txo.rs (hunk 4 above)
Comment (BrianCorbin) [reply full-service/src/db/txo.rs:1935]: Oh! hah. will update that :)

File: full-service/src/db/txo.rs (hunk 1 above)
Comment (eranrund) [reply full-service/src/db/txo.rs:1924]: Resolved over slack, this test checks for max spendable by the account and not by filtering for specific utxos that are under a certain value.

Comment (BrianCorbin) [reply full-service/src/db/txo.rs:1924]: update should have clarified this, but max spendable can have 2 meanings. In the context of the select_unspent_txos_for_value function, max_spendable_value is referring to the largest value an individual txo can have to be considered valid for the sake of selecting it for a transaction. In the context of the test, max_spendable_in_account is referring to "what is the maximum spendable amount of MOB that I can have be part of a single transaction"

**Comment by eranrund** [comment]
LGTM, thanks @BrianCorbin 
CI failure seems unrelated.

**Pull request merged**